### PR TITLE
Hotfix/4.2.8

### PR DESCRIPTION
**Pull request by ceo1647**
CHAM-217

### CODE REVIEW

File: src/Orc.Controls/Controls/LogMessageCategoryToggleButton/Themes/LogMessageCategoryToggleButton.generic.xaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                              Margin="0,0,4,0">
+                            <Grid.ColumnDefinitions>
+                                <ColumnDefinition Width="Auto" />
+                                <ColumnDefinition Width="*" />
+                                <ColumnDefinition Width="Auto" />
+                                <ColumnDefinition Width="Auto" />
+                            </Grid.ColumnDefinitions>
+
+                            <orccontrols:LogMessageCategoryControl Grid.Column="0" 
+                                                                   Category="{Binding RelativeSource={RelativeSource AncestorType={x:Type orccontrols:LogMessageCategoryToggleButton}}, Path= Category}" />
```
Comment (GeertvanHorrik) [inline src/Orc.Controls/Controls/LogMessageCategoryToggleButton/Themes/LogMessageCategoryToggleButton.generic.xaml:26]: Check space before Category

File: src/Orc.Controls/Extensions/ToggleButtonExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//   Copyright (c) 2008 - 2021 WildGums. All rights reserved.
+// </copyright>
+// --------------------------------------------------------------------------------------------------------------------
+
+
+namespace Orc.Controls
+{
+    using System.Windows.Controls.Primitives;
+
+    internal static class ToggleButtonExtensions
```
Comment (GeertvanHorrik) [inline src/Orc.Controls/Extensions/ToggleButtonExtensions.cs:12]: what's the added value of makin this internal?

File: src/Orc.Controls/Orc.Controls.csproj [hunk 3]
```diff
@@ -57,6 +57,24 @@
     <Folder Include="Controls\EditableControls\Behaviors\Base\" />
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (GeertvanHorrik) [inline src/Orc.Controls/Orc.Controls.csproj:60]: remove

File: src/Orc.Controls/Orc.Controls.csproj [hunk 4]
```diff
@@ -57,6 +57,24 @@
...
+    <Compile Update="Properties\Resources.Designer.cs">
+      <DesignTime>True</DesignTime>
+    </Compile>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Page Update="Controls\LogMessageCategoryToggleButton\Themes\LogMessageCategoryToggleButton.generic.xaml">
```
Comment (GeertvanHorrik) [inline src/Orc.Controls/Orc.Controls.csproj:67]: remove

File: src/Orc.Controls/Controls/LogMessageCategoryToggleButton/LogMessageCategoryToggleButton.cs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        public static readonly DependencyProperty CategoryProperty = DependencyProperty.Register(nameof(Category),
+            typeof(string), typeof(LogMessageCategoryToggleButton), new PropertyMetadata("",
+                (sender, args) => ((LogMessageCategoryToggleButton)sender).OnCategoryChanged(args)));
+        #endregion
+
+        public override void OnApplyTemplate()
+        {
+            base.OnApplyTemplate();
+
+            _toggleButton = (ToggleButton)GetTemplateChild("PART_Toggle");
```
[the remark this answers is not in the document]
Comment (ceo1647) [reply src/Orc.Controls/Controls/LogMessageCategoryToggleButton/LogMessageCategoryToggleButton.cs:54]: yes, that was an idea

**Review by GeertvanHorrik — APPROVED** [review]

**Pull request merged**